laske_export/document/invoice_sales_order_adapter.py

```diff
@@ -304,5 +319,133 @@ def set_values(self):
         self.set_dates()
         self.set_references()
 
-        line_items = self.get_line_items()
-        self.sales_order.line_items = line_items
+        self.sales_order.line_items = self.get_line_items()
+
+
+class AkvInvoiceSalesOrderAdapter(InvoiceSalesOrderAdapter):
+    """Adapter for service unit Alueiden kehittäminen ja valvonta.
+
+    For service-unit-aware instantiation of class objects, please use the
+    factory function invoice_sales_order_adapter_factory
+    """
+
+    AKV_DATE_FORMAT = "%d.%m.%Y"
+
+    def get_bill_text(self) -> str:
+        """Create bill text
+
+        AKV requested that "Hki otsikko ulkoinen" is left empty in their SAP.
+        "Hki otsikko ulkoinen" corresponds to XML elements BillTextL<number> in
+        the MVJ export.
+        """
+        number_of_bill_text_lines = 6
+        return "\n" * number_of_bill_text_lines
+
+    def get_line_items(self) -> list[LineItem]:
+        """Create line items for the AKV service unit."""
+        line_items: list[LineItem] = []
+
+        invoice_rows = self.invoice.rows.all()
+        for invoice_row in invoice_rows:
+            line_text = self.get_line_text(invoice_row)
+            text_lines = textwrap.wrap(
+                line_text, width=70, max_lines=6, drop_whitespace=False
+            )
+            item = LineItem()
+            for i in range(0, 6):
```

Comment:
This seems a bit mysterious in what is the purpose. What does the range refer to, could it be named?